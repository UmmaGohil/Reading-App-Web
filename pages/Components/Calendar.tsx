import Page from "../Pages/Page";
import { enGB } from "date-fns/locale";
import { DateChangeCallBack, DateRangePicker } from "react-nice-dates";
import "react-nice-dates/build/style.css";
import CSS from "csstype";

interface Props {
	// rome-ignore lint/suspicious/noExplicitAny: <explanation>
	startDate: Date | string | any;
	setStartDate: DateChangeCallBack;
	// rome-ignore lint/suspicious/noExplicitAny: <explanation>
	endDate: Date | string | any;
	setEndDate: DateChangeCallBack;
}

const input: CSS.Properties = {
	width: "7rem",
	height: "2rem",
	color: "#2938fa",
	border: "1px solid #f1a5d0",
	fontSize: "1rem",
	marginLeft: "1rem",
	borderRadius: "0.5rem",
	textAlign: "center",
};

const daterange: CSS.Properties = {
	marginTop: "-11vh",
};
const Calendar = ({ startDate, setStartDate, endDate, setEndDate }: Props) => {
	//const useIsomorphicLayoutEffect = typeof window !== "undefined" ? useLayoutEffect : useEffect;

	return (
		<Page variant="regular">
			<DateRangePicker
				startDate={startDate}
				endDate={endDate}
				onStartDateChange={setStartDate}
				onEndDateChange={setEndDate}
				minimumDate={new Date()}
				minimumLength={1}
				format="dd MMM yyyy"
				locale={enGB}
			>
				{({ startDateInputProps, endDateInputProps, focus }) => (
					<div style={daterange}>
						<input
							className={`input${focus === startDate ? " -focused" : ""}`}
							{...startDateInputProps}
							placeholder="Start date"
							style={input}
						/>
						<span className="date-range_arrow" />
						<input
							className={`input${focus === endDate ? " -focused" : ""}`}
							{...endDateInputProps}
							placeholder="End date"
							style={input}
						/>
					</div>
				)}
			</DateRangePicker>
		</Page>
	);
};

export default Calendar;
