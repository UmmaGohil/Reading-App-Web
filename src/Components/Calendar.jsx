import React, {useState} from 'react';
import Page from '../Pages/Page'
import { enGB } from 'date-fns/locale'
import { DateRangePicker, START_DATE, END_DATE } from 'react-nice-dates'
import 'react-nice-dates/build/style.css'

const Calendar = () => {
  const [startDate, setStartDate] = useState()
  const [endDate, setEndDate] = useState()

  // convert the dates and then computer the amount of days between them 
  // then send this down to results via a prop 

  console.log(startDate)
  console.log(endDate)

  const oneDay = 24 * 60 * 60 * 1000;

  const diff = Math.round(Math.abs((startDate - endDate/ (oneDay))))


  console.log(diff)

  return(
    <Page variant="regular">
      <DateRangePicker
      startDate={startDate}
      endDate={endDate}
      onStartDateChange={setStartDate}
      onEndDateChange={setEndDate}
      minimumDate={new Date()}
      minimumLength={1}
      format='dd MMM yyyy'
      locale={enGB}
    >
      {({ startDateInputProps, endDateInputProps, focus }) => (
        <div className='date-range'>
          <input
            className={'input' + (focus === START_DATE ? ' -focused' : '')}
            {...startDateInputProps}
            placeholder='Start date'
          />
          <span className='date-range_arrow' />
          <input
            className={'input' + (focus === END_DATE ? ' -focused' : '')}
            {...endDateInputProps}
            placeholder='End date'
          />
        </div>
      )}
    </DateRangePicker>
    
    </Page>
  )
}
export default Calendar

