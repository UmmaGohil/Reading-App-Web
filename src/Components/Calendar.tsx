import React from 'react';
import Page from '../Pages/Page'
import { enGB } from 'date-fns/locale'
import { DateRangePicker} from 'react-nice-dates'
import 'react-nice-dates/build/style.css'
import CSS from 'csstype'

interface Props {
  startDate: any,
  setStartDate:  any,
  endDate: any,
  setEndDate: any
}

const input: CSS.Properties = {
  width: '7rem',
  height: '2rem',
  color: '#2938fa',
  border: '1px solid #f1a5d0', 
  fontSize: '1rem',
  marginLeft: '1rem',
  borderRadius: '0.5rem',
  textAlign: 'center'
};

const Calendar = ({startDate, setStartDate, endDate, setEndDate}: Props) => {
  return(
    <Page variant="regular">
      <DateRangePicker
        startDate={startDate}
        endDate={endDate}
        onStartDateChange={setStartDate}
        onEndDateChange={setEndDate}
        minimumDate={new Date()}
        minimumLength={1}
        format='dd MMM yyyy'
        locale={enGB}
      >
      {({ startDateInputProps, endDateInputProps, focus }) => (
        <div className='date-range'>
          <input
            className={'input' + (focus === startDate ? ' -focused' : '')}
            {...startDateInputProps}
            placeholder='Start date'
            style={input}
          />
          <span className='date-range_arrow' />
          <input
            className={'input' + (focus === endDate ? ' -focused' : '')}
            {...endDateInputProps}
            placeholder='End date'
            style={input}
          />
        </div>
      )}
    </DateRangePicker>
    </Page>
  )
}

export default Calendar

