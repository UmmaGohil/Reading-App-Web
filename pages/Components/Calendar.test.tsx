import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import Calendar from "./Calendar";

it("renders correctly", () => {
	const result = render(
		<Calendar
			startDate={1 / 1 / 1}
			setStartDate={() => {}}
			endDate={1 / 1 / 1}
			setEndDate={() => {}}
		/>,
	);
	expect(result).toMatchSnapshot();
});
