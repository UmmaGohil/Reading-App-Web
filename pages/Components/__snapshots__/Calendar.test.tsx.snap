// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        style="color: rgb(41, 56, 250); background-color: rgb(241, 165, 208); font-size: 5rem;"
      />
      <div
        class="nice-dates"
      >
        <div
          style="margin-top: -11vh;"
        >
          <input
            class="input"
            placeholder="Start date"
            style="width: 7rem; height: 2rem; color: rgb(41, 56, 250); border: 1px solid #f1a5d0; font-size: 1rem; margin-left: 1rem; border-radius: 0.5rem; text-align: center;"
            type="text"
            value=""
          />
          <span
            class="date-range_arrow"
          />
          <input
            class="input"
            placeholder="End date"
            style="width: 7rem; height: 2rem; color: rgb(41, 56, 250); border: 1px solid #f1a5d0; font-size: 1rem; margin-left: 1rem; border-radius: 0.5rem; text-align: center;"
            type="text"
            value=""
          />
        </div>
        <div
          class="nice-dates-popover"
        >
          <div>
            <div
              class="nice-dates-navigation"
            >
              <a
                class="nice-dates-navigation_previous"
              />
              <span
                class="nice-dates-navigation_current"
              >
                January 1970
              </span>
              <a
                class="nice-dates-navigation_next"
              />
            </div>
            <div
              class="nice-dates-week-header"
            >
              <span
                class="nice-dates-week-header_day"
              >
                Mon
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Tue
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Wed
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Thu
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Fri
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Sat
              </span>
              <span
                class="nice-dates-week-header_day"
              >
                Sun
              </span>
            </div>
            <div
              class="nice-dates-grid"
              style="height: 6px;"
            >
              <div
                class="nice-dates-grid_container -origin-top"
                style="transform: translate3d(0, 0px, 0); transition-duration: 500ms;"
              >
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    29
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    30
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    31
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_month"
                  >
                    Jan
                  </span>
                  <span
                    class="nice-dates-day_date"
                  >
                    1
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    2
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    3
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    4
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    5
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    6
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    7
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    8
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    9
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    10
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    11
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    12
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    13
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    14
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    15
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    16
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    17
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    18
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    19
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    20
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    21
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    22
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    23
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    24
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    25
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    26
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    27
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    28
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    29
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    30
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    31
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_month"
                  >
                    Feb
                  </span>
                  <span
                    class="nice-dates-day_date"
                  >
                    1
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    2
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    3
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    4
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    5
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    6
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    7
                  </span>
                </span>
                <span
                  class="nice-dates-day -disabled -outside"
                  style="height: 1px;"
                >
                  <span
                    class="nice-dates-day_date"
                  >
                    8
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      style="color: rgb(41, 56, 250); background-color: rgb(241, 165, 208); font-size: 5rem;"
    />
    <div
      class="nice-dates"
    >
      <div
        style="margin-top: -11vh;"
      >
        <input
          class="input"
          placeholder="Start date"
          style="width: 7rem; height: 2rem; color: rgb(41, 56, 250); border: 1px solid #f1a5d0; font-size: 1rem; margin-left: 1rem; border-radius: 0.5rem; text-align: center;"
          type="text"
          value=""
        />
        <span
          class="date-range_arrow"
        />
        <input
          class="input"
          placeholder="End date"
          style="width: 7rem; height: 2rem; color: rgb(41, 56, 250); border: 1px solid #f1a5d0; font-size: 1rem; margin-left: 1rem; border-radius: 0.5rem; text-align: center;"
          type="text"
          value=""
        />
      </div>
      <div
        class="nice-dates-popover"
      >
        <div>
          <div
            class="nice-dates-navigation"
          >
            <a
              class="nice-dates-navigation_previous"
            />
            <span
              class="nice-dates-navigation_current"
            >
              January 1970
            </span>
            <a
              class="nice-dates-navigation_next"
            />
          </div>
          <div
            class="nice-dates-week-header"
          >
            <span
              class="nice-dates-week-header_day"
            >
              Mon
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Tue
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Wed
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Thu
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Fri
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Sat
            </span>
            <span
              class="nice-dates-week-header_day"
            >
              Sun
            </span>
          </div>
          <div
            class="nice-dates-grid"
            style="height: 6px;"
          >
            <div
              class="nice-dates-grid_container -origin-top"
              style="transform: translate3d(0, 0px, 0); transition-duration: 500ms;"
            >
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  29
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  30
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  31
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_month"
                >
                  Jan
                </span>
                <span
                  class="nice-dates-day_date"
                >
                  1
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  2
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  3
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  4
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  5
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  6
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  7
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  8
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  9
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  10
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  11
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  12
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  13
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  14
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  15
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  16
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  17
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  18
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  19
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  20
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  21
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  22
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  23
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  24
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  25
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  26
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  27
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  28
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  29
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  30
                </span>
              </span>
              <span
                class="nice-dates-day -disabled"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  31
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_month"
                >
                  Feb
                </span>
                <span
                  class="nice-dates-day_date"
                >
                  1
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  2
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  3
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  4
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  5
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  6
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  7
                </span>
              </span>
              <span
                class="nice-dates-day -disabled -outside"
                style="height: 1px;"
              >
                <span
                  class="nice-dates-day_date"
                >
                  8
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
